libs/langchain/ragstack_langchain/graph_store/extractors/edge_extractor.py

```diff
@@ -1,28 +1,25 @@
 from __future__ import annotations
 
 from abc import ABC, abstractmethod
-from typing import Generic, Iterable, Iterator, Set, TypeVar
+from typing import Generic, Iterable, TypeVar
 
 from langchain_core.documents import Document
 from ragstack_knowledge_store._utils import strict_zip
-from ragstack_knowledge_store.link_tag import LinkTag
 
 InputT = TypeVar("InputT")
 
 
 class EdgeExtractor(ABC, Generic[InputT]):
     @abstractmethod
-    def extract_one(self, document: Document, input: InputT):
+    def extract_one(self, document: Document, input: InputT) -> None:
```

Comment:
I wonder if it would be cleaner for this to not take `Document` and return the `Set[Link]` instead. Then we could allow: `add_links(document, link1, link2, link3)` or `add_links(document, *edge_extractor(input))`, etc.

Reply:
I also prefer not mutating args when the function name is not clearly explicit about it.

Reply:
Yeah. I think I like that. We could change

```
def add_links(doc_or_md, *links: Link) -> None:
  ...
```

to
```
def add_links(doc_or_md, *links: Union[Link, Iterable[Link]]) -> None:
  ...
```

To make it even easier. Then we can write `add_links(document, edge_extractor(input), edge_extractor2(input))` and it will automatically add all the edges to the document or metadata. And the `add_links` becomes the only thing that mutates the document.

Reply:
Let's o another PR for that.